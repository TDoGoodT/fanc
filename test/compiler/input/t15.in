int multiply(int a, int bb) {
    return a * bb;
}
int decrement(int a) {
    return a - 1;
}
int increment(int a) {
    return a + 1;
}
int subtract(int a, int ab) {
    int tmpa = a;
    int tmpb = ab;
    while (tmpb > 0) {
        tmpa = decrement(tmpa);
        tmpb = decrement(tmpb);
    }
    return tmpa;
}
int modulo(int a, int ab) {
    int tmpa = a;
    while (tmpa >= ab) {
        tmpa = subtract(tmpa, ab);
    }
    return tmpa;
}
int gcd(int a, int c) {
    int tmpa = a;
    int tmpc = c;
    while (tmpa != tmpc) {
        if (tmpa > tmpc) {
            tmpa = subtract(tmpa, tmpc);
        } else {
            tmpc = subtract(tmpc, tmpa);
        }
    }
    return tmpa;

}

void main() {
    int num1 = 15;
    int num2 = 20;

    int result = gcd(num1, num2);

    print("GCD of 15 and 20: ");
    printi(result);
}
