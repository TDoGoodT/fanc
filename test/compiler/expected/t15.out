GCD of 15 and 20: 
5
